### ISSUE TITLE
[promptehr-pr-prep] T8: Write PromptEHR integration tests

### ISSUE THREAD

**Issue by jalengg** [jalengg/PyHealth#53]
## Task

Create `tests/integration/test_promptehr_end_to_end.py` with 8 MECE in-memory tests + 4 MIMIC-III skip-gracefully tests.

## The 8 MECE Category A Tests

1. `TestPromptEHRIsBaseModelInstance` — `isinstance(model, BaseModel)`
2. `TestPromptEHRFeatureKeys` — `model.feature_keys == ["visits"]`
3. `TestPromptEHRVocabSize` — input dim matches `dataset.input_processors["visits"].vocab_size()`
4. `TestPromptEHRForwardRaisesNotImplementedError` — `forward()` raises `NotImplementedError`
5. `TestPromptEHRTrainModelRuns` — 1 epoch completes without exception
6. `TestPromptEHRSynthesizeCount` — returns correct number of records
7. `TestPromptEHRSynthesizeOutputStructure` — nested list of strings, not flat
8. `TestPromptEHRSaveLoadRoundtrip` — checkpoint persists and loads correctly

## Bootstrap Pattern

`transformers` IS installed in `/u/jalenj4/pehr_scratch/venv` — no stubbing needed. Use the same `_bootstrap()` as HALO/CorGAN but stub only `litdata`, `pyarrow`, and `pyhealth.models.__init__`.

Small test dataset (minimum 2 visits per patient, 8 patients for batch_size=4):
```python
_SMALL_SAMPLES = [
    {"patient_id": "p1", "visits": [["428", "250"], ["401", "414"]], "age": 65.0, "gender": "M"},
    {"patient_id": "p2", "visits": [["250", "401"], ["428"]], "age": 55.0, "gender": "F"},
    # ... 8 total
]
_SMALL_MODEL_KWARGS = dict(epochs=1, batch_size=4, save_dir=None)
```

## Why Tests Are Critical

PromptEHR uses HuggingFace `Trainer` — a significantly more complex training infrastructure than HALO/CorGAN. Tests must verify the HuggingFace Trainer integrates correctly with PyHealth's SampleDataset, that generation returns PyHealth-standard output (not HuggingFace token sequences), and that the BaseModel contract is fulfilled despite the internal complexity.

## 4 Category B MIMIC-III Tests (skip-gracefully)

1. `test_mimic3_set_task_returns_nonempty_dataset`
2. `test_mimic3_sample_keys` — check `patient_id`, `visits`, `age`, `gender` present
3. `test_mimic3_visits_are_nested_sequence_tensors`
4. `test_mimic3_full_pipeline_train_and_synthesize`

**Comment by jalengg** [comment]
T8 complete. Commit: ad0b27b72a75e41dbfaec7f5d28f2bf141747c0a. 8 Category A tests pass, 4 Category B MIMIC-III tests skip gracefully.

**Comment by jalengg** [comment]
## T8 Implementation Notes

**Commit**: `ad0b27b7`

### What was implemented

Created `tests/integration/test_promptehr_end_to_end.py` with 12 tests: 8 Category A (in-memory, always pass) + 4 Category B (MIMIC-III, skip gracefully).

**Category A tests:**
1. `TestPromptEHRIsBaseModelInstance` — `isinstance(model, BaseModel)`
2. `TestPromptEHRFeatureKeys` — `model.feature_keys == ["visits"]`
3. `TestPromptEHRVocabSize` — `model._vocab.total_size == processor.vocab_size() + 5`
4. `TestPromptEHRForwardRaisesNotImplementedError`
5. `TestPromptEHRTrainModelRuns` — 1 epoch, tiny model; verifies checkpoint file is saved
6. `TestPromptEHRSynthesizeCount` — `synthesize_dataset(3)` returns exactly 3 dicts
7. `TestPromptEHRSynthesizeOutputStructure` — each dict has `patient_id` (str) and `visits` (nested list of str)
8. `TestPromptEHRSaveLoadRoundtrip` — save → load → synthesize still works

**Category B tests** (skip if MIMIC-III absent):
- `test_mimic3_set_task_returns_nonempty_dataset`
- `test_mimic3_sample_keys` — each sample has `patient_id`, `visits`, `age`, `gender`
- `test_mimic3_visits_are_nested_tensors` — `NestedSequenceProcessor` output is list of `torch.long` tensors
- `test_mimic3_full_pipeline_train_and_synthesize` — 1-epoch train + synthesize 5 patients

### Key design decisions

**Bootstrap pattern** — same as `test_corgan_end_to_end.py`: stub `pyhealth.datasets` and `pyhealth.models` as `MagicMock`, load `base_model.py` and each `promptehr` submodule directly via `importlib.util.spec_from_file_location`, stub `litdata` for `sample_dataset.py`. This avoids triggering heavy optional deps (`einops`, `litdata`, `pyarrow`) that are absent from the venv.

**`_TINY_BART_CONFIG`** — Tests use a `BartConfig(d_model=32, encoder_layers=1, decoder_layers=1, ...)` instead of `facebook/bart-base` (768-dim, 6 layers). This required adding support for passing a `BartConfig` object directly as `bart_config_name` (a backward-compatible change to `model.py`). Without this, tests would take 10+ minutes and require internet access.

**`train_model` before `synthesize_dataset`** — Unlike CorGAN (which can synthesize from a randomly initialized GAN), PromptEHR's autoregressive BART `generate()` requires a properly initialized model with a populated `_demo_pool`. Tests 6, 7, and 8 use `setUpClass` to train once and reuse the model, keeping total test time under 2 minutes.

**`warmup_steps=0`** — Set to 0 in test model kwargs to avoid the HuggingFace scheduler warning about warmup steps exceeding training steps when using only 1 epoch with a tiny dataset.

**Result**: 8 passed, 4 skipped in 1m 54s on CPU.
